### ISSUE TITLE
Documentation lacking for a lot of patterns

### ISSUE THREAD

**Issue by janga1997** [Patternslib/Patterns#457]
Documentation and demos for a lot of patterns is missing. I would be willing to work on it, except I have to individually figure out how each of them works.
Any views @jcbrand ?

**Comment by jcbrand** [comment]
Yes, the documentation is lacking in many cases, that's true. If you're willing to do the work of writing documentation, which would be much appreciated, then I'll be happy to help you by answering any questions you might have or giving feedback on what you write.

If you don't know this already, the demo and documentation for each pattern is contained inside the same directory as it's code and tests.

For exaple: https://github.com/Patternslib/Patterns/blob/master/src/pat/auto-suggest/

The documentation is in `documentation.md` and the example is in `index.html`

**Comment by cornae** [comment]
I will be happy to help too. In general @jcbrand knows all about the JavaScript side and I would be able to answer more questions on the HTML/CSS side of things and Patterns syntax and design issues.
